doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -520,8 +647,38 @@ Sections :ref:`if-then-else` and :ref:`irrefutable-patterns`).
    single: fix
    single: cofix
 
-Recursive functions
--------------------
+Recursive and co-recursive functions: fix and cofix
+---------------------------------------------------
+
+.. insertgram term_fix term1_or_global_opt
+
+.. productionlist:: coq
+   term_fix : `single_fix`
+            : `single_fix` with `fix_body_list` for `ident`
+   single_fix : fix `fix_body`
+              : cofix `fix_body`
+   fix_body_list : `fix_body_list` with `fix_body`
+                 : `fix_body`
+   fix_body : `ident` `binders_fixannot` `colon_term_opt` := `term`
+   binders_fixannot : { `ident` `impl_ident` } `binders_fixannot`
+                    : `fixannot`
+                    : `binder` `binders_fixannot`
+                    : `empty`
+   impl_ident : `name_list` : `term`
+              : `name_list`
+              : : `term`
+              : `empty`
+   fixannot : { struct `ident` }
+            : { wf `term1_or_global` `ident` }
+            : { measure `term1_or_global` `ident_opt` `term1_or_global_opt` }
+            : { struct `name` }
```

Comment:
Note that this `fixannot` non-terminal could be reused in Chapter "Extensions of Gallina" instead of `decrease_annot`, which is redefined there.

Reply:
Let me fix `fixannot` first.